### PR TITLE
Change to ruff

### PR DESCRIPTION
**Pull request by FoamyGuy**
I recently acquired the hardware that this driver is for and got set up locally to test a different PR, so I've gone ahead with the changes needed for switching to ruff.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# SPDX-FileCopyrightText: 2024 Justin Myers for Adafruit Industries
+#
+# SPDX-License-Identifier: Unlicense
+
+* text eol=lf
```
Comment (tannewt) [inline .gitattributes]: I suspect this is too aggressive and will break any images. Maybe limit this to py, rst, txt, yaml and toml?

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for the update! One comment about gitattributes.

File: .gitattributes (hunk 1 above)
Comment (FoamyGuy) [reply .gitattributes]: I've added a commit that specifies those file extensions. I don't have much experience with the .gitattributes file syntax. I looked into the docs a bit and I believe I've done it in a way that is valid, but am not sure if there is a better way than listing them all out though. If so, I'm happy to update it however. 

I'll go back and submit PRs to update the `.gitattributes` files in the other repos that have been done already as well once this one is set.

Comment (FoamyGuy) [reply .gitattributes]: I also added two more to the list you mentioned:
```
.license
.md
```
There are plain text files with both of those extensions checked into the repo as well. I can remove them if they're unneeded in this list though.

**Review by tannewt — APPROVED** [review]
Much better! Thank you!

**Pull request merged**